include/wil/registry_basic.h

```diff
@@ -38,1865 +36,1779 @@
 
-    inline void set_value_qword(HKEY key, _In_opt_ PCWSTR value_name, uint64_t data)
-    {
-        return set_value_qword(key, nullptr, value_name, data);
-    }
+        inline void set_value_expanded_string(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, _In_ PCWSTR data) WI_NOEXCEPT
+        {
+            const reg_view_details::reg_view regview{ key };
+            return regview.set_value_with_type(subkey, value_name, data, REG_EXPAND_SZ);
+        }
 
-    inline void set_value_string(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, _In_ PCWSTR data)
-    {
-        const reg_view_details::reg_view regview{ key };
-        return regview.set_value(subkey, value_name, data);
-    }
+        inline void set_value_expanded_string(HKEY key, _In_opt_ PCWSTR value_name, _In_ PCWSTR data) WI_NOEXCEPT
+        {
+            return ::wil::reg::set_value_expanded_string(key, nullptr, value_name, data);
+        }
 
-    inline void set_value_string(HKEY key, _In_opt_ PCWSTR value_name, _In_ PCWSTR data)
-    {
-        return set_value_string(key, nullptr, value_name, data);
-    }
 #endif // #if defined(WIL_ENABLE_EXCEPTIONS)
 
 #if defined(_VECTOR_) && defined(_STRING_) && defined(WIL_ENABLE_EXCEPTIONS)
-    inline void set_value_multistring(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, const ::std::vector<::std::wstring>& data)
-    {
-        const reg_view_details::reg_view regview{ key };
-        return regview.set_value_multistring(subkey, value_name, data);
-    }
+        inline void set_value_multistring(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, const ::std::vector<::std::wstring>& data)
+        {
+            const reg_view_details::reg_view regview{ key };
+            return regview.set_value_multistring(subkey, value_name, data);
+        }
 
-    inline void set_value_multistring(HKEY key, _In_opt_ PCWSTR value_name, const ::std::vector<::std::wstring>& data)
-    {
-        return set_value_multistring(key, nullptr, value_name, data);
-    }
+        inline void set_value_multistring(HKEY key, _In_opt_ PCWSTR value_name, const ::std::vector<::std::wstring>& data)
+        {
+            return ::wil::reg::set_value_multistring(key, nullptr, value_name, data);
+        }
 
-    template <>
-    inline void set_value(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, const ::std::vector<::std::wstring>& data)
-    {
-        return set_value_multistring(key, subkey, value_name, data);
-    }
-    template <>
-    inline void set_value(HKEY key, _In_opt_ PCWSTR value_name, const ::std::vector<::std::wstring>& data)
-    {
-        return set_value_multistring(key, nullptr, value_name, data);
-    }
+        template <>
+        inline void set_value(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, const ::std::vector<::std::wstring>& data)
+        {
+            return ::wil::reg::set_value_multistring(key, subkey, value_name, data);

```

Comment:
Does this mean `reg_view`'s `set_value` won't work for `set_value<std::vector<std::wstring>>`?

Reply:
right. not without reworking a bunch of stuff.